### PR TITLE
gdu: 2.3.0 -> 3.0.0

### PR DESCRIPTION
**Pull request by zowoq**
https://github.com/dundee/gdu/releases/tag/v3.0.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 109491` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gdu</li>
  </ul>
</details>

File: pkgs/tools/system/gdu/default.nix [hunk 1]
```diff
@@ -6,21 +6,21 @@
...
-  vendorSha256 = "18a3qwshz8jmw0j29qvmzarxig0kj1n0fnmlx81qzswsyl85kncv";
+  vendorSha256 = "0w3k23kly8g9mf8a300xz6bv7g1m2nlp5f112k4viyi9zy6vqbv0";
 
-  buildFlagsArray = [ "-ldflags=-s -w -X main.AppVersion=${version}" ];
+  buildFlagsArray = [ "-ldflags=-s -w -X github.com/dundee/gdu/build.Version=${version}" ];
 
-  # analyze/dev_test.go: undefined: processMounts
-  doCheck = !stdenv.isDarwin;
+  # tests fail is the version is set
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/tools/system/gdu/default.nix:23]: `TestVersion` is using `development`. Looks like that the tests could pass by replacing the `development` with the current version.

```suggestion
  postPatch = ''
    substituteInPlace cmd/cmd_test.go --replace "development" "${version}"
  '';
```

**Review by fabaff — COMMENTED** [review]
- [x] package build (see https://github.com/NixOS/nixpkgs/pull/109491#issuecomment-761532391)
- [ ] all depending packages build
- [x] executables tested on x86_64

```bash
w/pr-109491/results]$ ./gdu/bin/gdu -v
Version:	 3.0.0
Built time:	 
Built user:	 
```

I'm not sure about the guidelines in regard of additional details in the output which are set by the build system. Perhaps we could set the time to the current day or add a timestamp and set `User` to `Nix Build System` or so.

Otherwise, looks good to me.

**Comment by zowoq** [comment]
> Perhaps we could set the time to the current day or add a timestamp and set User to Nix Build System or so.

Time would need to be fixed value (i.e. SOURCE_DATE_EPOCH or similar) otherwise it's an impurity.

As neither of them are necessary or provide useful information I don't see that we need to set them.

> Looks like that the tests could pass by replacing `development` with the current version.

I don't think we should bother with this as it doesn't seem like it is meant to be run by downstream packagers and the tests still don't work on some platforms anyway. If the quick releases by upstream continue we'd likely need to keep adjusting it as well.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109491` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>gdu</li>
</ul>
</details>

**Pull request merged**